platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -913,7 +914,10 @@ private void initialize(@NonNull final Context context, @NonNull Style style,
     if (isComponentInitialized) {
       return;
     }
-    isComponentInitialized = true;
+
+    if (!style.isFullyLoaded()) {
+      throw new IllegalStateException("Style hasn't fully loaded yet, can't initialize LocationComponent.");
```

Comment:
This scenario will only happen if somebody is using an old style, so the message should be more along the lines of "Style is invalid, provide a most recently loaded one."